app/src/App/Controller/Administrator/SendActivationEmail.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Controller\Administrator;
+
+use Demo\Domain\Model\Administrator\AdministratorRepository;
+use Demo\Domain\Service\Administrator\SendActivationEmailInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class SendActivationEmail
+{
+    public function __construct(
+        private SendActivationEmailInterface $sendActivationEmail,
+        private AdministratorRepository $administratorRepository
+    ) {
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $response = new Response();
+
+        $administratorId = (int) $request->attributes->get('id');
+
+        $administrator = $this->administratorRepository->find($administratorId);
+        if ($administrator === null) {
+            throw new \DomainException("Resource not found", 404);
+        }
+
+        try {
+            $this->sendActivationEmail->execute($administrator);
```

Comment:
This is technically an application service but we can keep it like this becasuse it's so small